.github/workflows/build-test.yml

```diff
@@ -25,19 +31,30 @@ jobs:
 
       - name: Build
         id: build
-        run: mvn -B -U -Dsurefire.rerunFailingTestsCount=5 clean install
+        run: mvn -B -U "-Dsurefire.rerunFailingTestsCount=5" clean install
 
       - name: Archive Test Results on Failure
         uses: actions/upload-artifact@v2
-        if: failure()
+        if: always()
         with:
-          name: test-results
+          name: Unit Test Results ${{ matrix.os }}
           path: target/surefire-reports/
           retention-days: 7
 
-      - name: Publish Unit Test Results
-        id: publish
-        uses: EnricoMi/publish-unit-test-result-action@v1.26
-        if: failure()
-        with:
-          files: target/surefire-reports/*.xml
+  publish-test-results:
+      name: Publish Unit Test Results
+      needs: build-and-test
```

Comment:
🔧  I was a bit confused about the needs here, because that requires the referred job to have succeeded. It would be nice to have it directly next to the `if: always()`